numba/tests/test_dictobject.py

```diff
@@ -1229,6 +1229,22 @@ def foo(d):
         # Value is correctly updated
         self.assertPreciseEqual(d[1], np.arange(10, dtype=np.int64) + 100)
 
+    def test_storage_model_mismatch(self):
```

Comment:
These tests need a python 2 guard as they use the string type.

Reply:
done